UI/context-bar-controls.cpp

```diff
@@ -33,6 +34,67 @@ SourceToolbar::SourceToolbar(QWidget *parent, OBSSource source)
 {
 }
 
+SourceToolbar::~SourceToolbar()
+{
+	if (oldData)
+		obs_data_release(oldData);
+}
```

Comment:
The point of using an OBSData object is to prevent having to do any sort of manual release. This code should not be necessary.

When using an OBSData object, you should instead immediately call `obs_data_release()` after core libobs function calls that return an incremented reference. So for example, if you do, say, `oldData = obs_data_create();`, you'd want to call `obs_data_release(oldData);` right after because obs_data_create() increments the reference. That way, the final release is handled by RAII and you won't need to do any sort of extra manual release later.

Here's an existing example of this: https://github.com/obsproject/obs-studio/blob/8f3d4b6758844b6c67574e95b55f2fe52ed4cc65/UI/source-tree.cpp#L525-L526

To be honest I just need to implement a method to pass an incremented value from core libobs functions so this confusion doesn't happen again in the future and calling `obs_data_release()` is never necessary at all. I think I'm going to implement something for that.